### ISSUE TITLE
Find out how certain jorje classes are used

### ISSUE THREAD

**Issue by dangmai** [dangmai/prettier-plugin-apex#7]
For an application like `prettier-plugin-apex`, we need to be very thorough about handling all the possible AST elements that exist. However, there are a few jorje classes whose usage I cannot figure out. These include:

- In `apex.jorje.data.sosl.ReturningSelectExpr`, the variable `bind` with type `apex.jorje.data.soql.BindClause`. I've seen and handled `BindClause` being used in SOQL before, but I can't create a SOSL test code that contains it. [TODO location](https://github.com/dangmai/prettier-plugin-apex/blob/5b43e01764b412f316ba4e15dd08aa9631261639/src/printer.js#L1205)
- In `apex.jorje.data.sosl.Search`, the variable `using` with type `apex.jorje.data.sosl.SearchUsingClause`. Same as above, I can't come up with a SOST test code that contains it. [TODO location](https://github.com/dangmai/prettier-plugin-apex/blob/5b43e01764b412f316ba4e15dd08aa9631261639/src/printer.js#L1222)
- `apex.jorje.data.soql.UsingExpr.UsingId` - same as above, I can't write a SOQL or SOSL statement that contains it. [TODO location](https://github.com/dangmai/prettier-plugin-apex/blob/5b43e01764b412f316ba4e15dd08aa9631261639/src/printer.js#L1698)

Hoping that someone from Salesforce (wink wink @ntotten) can help with these.

**Comment by ntotten** [comment]
I'll ask somebody on the team to take a look and get back to you.

**Comment by vazexqi** [comment]
Here are some examples on how to get those nodes to be built. These examples are taken from our unit tests so they aren't full snippets since we only test those particular nodes in unit tests.

`ReturningSelectExpr`

```
(f1(f2(foo)), bar
(f1(f2(foo)), bar USING LOOKUP l WHERE i = 'baz' ORDER BY j LIMIT 8128 OFFSET 628 BIND i = 'foobar')
```

`SearchUsingClause `

```
USING PHRASE SEARCH
USING ADVANCED SEARCH
```

`UsingId`

```
table USING listView(id=someView)
```

**Comment by dangmai** [comment]
Thanks for your help @vazexqi @ntotten, I'll check out your examples and see if I can replicate them in my test cases.

**Comment by dangmai** [comment]
Fixed in 5d8c1aa8ca18d82fbed9da9404a9d78daff506a3, 7ae41fe0fea896ceaa8da325b292e7d817a566a7 and ef9e53abb401e65dcd171d19d2ab94134dbecfb6

Thank you for your pointers! I definitely wouldn't have been able to figure these node usage out without you guys. Even with your help I still haven't been able to construct a SOQL/SOSL query that would run with the bind clause yet, but at least I managed to create a syntactically correct SOQL/SOSL with it for testing purpose. I have to wonder why these are not mentioned at all in the [SOQL](https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select.htm) and [SOSL](https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_sosl_syntax.htm) syntax pages?
